community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/index/storage/FailureStorage.java

```diff
@@ -115,6 +115,10 @@ public synchronized void storeIndexFailure( String failure ) throws IOException
         File failureFile = failureFile();
         try ( StoreChannel channel = fs.open( failureFile, "rw" ) )
         {
+            byte[] existingData = new byte[(int) channel.size()];
+            channel.read( ByteBuffer.wrap( existingData ) );
+            channel.position( lengthOf( existingData ) );
```

Comment:
Why do you need to read the existing data? Can't you just `channel.position( channel.size() )`? Also, the read might not read all of the data in the first go. You also have a `channel.close` further down in this method.